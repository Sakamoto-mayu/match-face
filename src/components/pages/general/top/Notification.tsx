import { twMerge } from 'tailwind-merge';

type MenuProps = {
  title: string;
  completedTasksNumbers: number;
  incompletedTasksNumbers: number;
  deadlineDay: string;
  deadlineNumbers: number;
  className: string;
};

export default function Notification({
  title,
  completedTasksNumbers,
  incompletedTasksNumbers,
  deadlineDay,
  deadlineNumbers,
  className,
}: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const mergedClassName = twMerge('p-5 w-1/2 h-auto shadow-lg', className);
  return (
    <>
      <div className={mergedClassName}>
        <h1 className="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12">
          {title}
        </h1>

        <div className="">
          {/* answer_statusがtrueの件数? */}
          <h2 className="">【対応済】{completedTasksNumbers}件</h2>
          {/* answer_statusがfalseの件数? */}
          <h2 className="pt-4">【未対応】{incompletedTasksNumbers}件</h2>
          {/* データ構造がはっきりしてから編集要*/}
          <ul className=" pl-4">
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
          </ul>
        </div>
      </div>
    </>
  );
}
