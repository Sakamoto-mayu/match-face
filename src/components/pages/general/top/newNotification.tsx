import { twMerge } from 'tailwind-merge';

type MenuProps = {
  completedTasksNumbers: number;
  incompletedTasksNumbers: number;
  deadlineDay: string;
  deadlineNumbers: number;
  className: string;
};

export default function Notification({
  completedTasksNumbers,
  incompletedTasksNumbers,
  deadlineDay,
  deadlineNumbers,
  className,
}: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const mergedClassName = twMerge('p-5 w-3/5 h-auto shadow-lg', className);
  return (
    <>
      <div className={mergedClassName}>
        <h1 className="border-b-[2px] border-b-black shadow-md text-2xl text-left w-full font-bold h-12">
          進捗状況
        </h1>

        <div className="pt-5">
          <h2 className="">【対応済】{completedTasksNumbers}件</h2>
          <h2 className="pt-5">【未対応】{incompletedTasksNumbers}件</h2>
          <ul className=" pl-5">
            <li className="pt-2">
              {deadlineDay}までに対応必須の項目が{deadlineNumbers}件あります
            </li>
          </ul>
        </div>
      </div>
    </>
  );
}
