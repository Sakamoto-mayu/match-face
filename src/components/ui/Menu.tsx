import Image from 'next/image';
import Link from 'next/link';
import { ComponentProps } from 'react';
import { twMerge } from 'tailwind-merge';

type MenuProps = {
  link: {
    url: string;
  };
  elements: {
    title: string;
    description: string;
    imgUrl: string;
    imgAlt: string;
  };
} & ComponentProps<'button'>;

export default function Menu({ link, elements, ...props }: MenuProps) {
  //buttonのclassNameは親コンポーネントでclassNameを指定して上書き可能
  const className = twMerge(
    'flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200',
    props.className
  );

  return (
    <Link href={link.url}>
      <button {...props} className={className}>
        <span className="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold">
          {elements.title}
        </span>

        <div className="flex pt-4 w-full">
          <span className="text-left mr-auto">{elements.description}</span>
          <Image
            src={elements.imgUrl}
            alt={elements.imgAlt}
            width="70"
            height="70"
            className="pt-2 pl-2"
          />
        </div>
      </button>
    </Link>
  );
}
