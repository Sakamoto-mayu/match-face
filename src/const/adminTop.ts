export const dataArray = [
    {
      title: "質問・回答の追加・編集",
      url: "/admin/handle-question",
      description: "質問・回答例を追加・編集します。",
      imgUrl: "/icon/document_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "タスクの設定",
      url: "/admin/tasks/register",
      description: "ユーザーを選択してタスクを設定します。",
      imgUrl: "/icon/pencil_icon.png",
      imgAlt: "pencil_icon",
    },
    {
      title: "回答の閲覧",
      url: "/admin/histories",
      description: "ユーザーの回答を閲覧できます。",
      imgUrl: "/icon/search_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "ユーザー管理",
      url: "/admin/users",
      description: "ユーザーの一覧を確認できます。",
      imgUrl: "/icon/user_icon.png",
      imgAlt: "アイコン",
    },
    {
      title: "グループ設定",
      url: "/admin/groups/register",
      description: "複数人のユーザーから成るグループを設定します。",
      imgUrl: "/icon/group_icon.png",
      imgAlt: "アイコン",
    },
  ];
