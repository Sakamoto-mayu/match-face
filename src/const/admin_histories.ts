export const skills = [
    {id:1,skill:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,skill:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,skill:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,skill:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,skill:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,skill:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,skill:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,skill:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,skill:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,skill:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,skill:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,skill:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,skill:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,skill:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const projects = [
    {id:1,name:"案件A",detail:"案件Aの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"案件B",detail:"案件Bの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"案件C",detail:"案件Cの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"案件D",detail:"案件Dの説明",enterprise_id:1,department_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"案件E",detail:"案件Eの説明",enterprise_id:1,department_id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]

export const project_skills = [
    {id:1,project_id:1,skill_id:[1,2,3],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,project_id:2,skill_id:[4,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,project_id:3,skill_id:[8,13],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,project_id:4,skill_id:[1,2,5],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,project_id:5,skill_id:[2,3,4,6],created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answer_requests = [
    {id:1,user_id:[1,2],administrator_id:5,project_id:1,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,user_id:[1,2],administrator_id:5,project_id:2,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,user_id:[3],administrator_id:5,project_id:3,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,user_id:[1,2,4],administrator_id:5,project_id:4,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-07-15T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,user_id:[4],administrator_id:5,project_id:5,request_at:'2023-07-10T16:52:46.053Z',deadline:"2023-08-10T16:52:46.053Z",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const answers = [
    {id:1,context:'回答が入ります1',question_id:1,answer_request_id:1,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,context:'回答が入ります2',question_id:1,answer_request_id:1,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:3,context:'回答が入ります3',question_id:2,answer_request_id:2,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-13T16:52:46.053Z'},
    {id:4,context:'回答が入ります4',question_id:2,answer_request_id:2,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,context:'回答が入ります5',question_id:3,answer_request_id:3,user_id:3,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,context:'回答が入ります6',question_id:4,answer_request_id:4,user_id:1,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:7,context:'回答が入ります7',question_id:4,answer_request_id:4,user_id:2,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
    {id:8,context:'回答が入ります8',question_id:4,answer_request_id:4,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,context:'回答が入ります9',question_id:5,answer_request_id:5,user_id:4,Model_answer_fl:false,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-11T16:52:46.053Z'},
]

export const users = [
    {id:1,name:'田中太郎',password:'tanaka',email:'tanaka@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:'山田次郎',password:'yamada',email:'yamada@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:'佐藤三郎',password:'sato',email:'sato@example.com',hire_date:'2023-01-01',department_id:2,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:'鈴木四郎',password:'suzuki',email:'suzuki@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:'高橋五郎',password:'takahashi',email:'takahashi@example.com',hire_date:'2023-01-01',department_id:1,status_id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]
