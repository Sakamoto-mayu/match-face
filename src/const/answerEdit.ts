const answerEditData = [
  {
    id: 1,
    project_name: "test1",
    project_detail:
      "案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示案件詳細が表示",
    questions: [
      {
        question_id: 1,
        question:
          "問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１問題文１",
        answer_example:
          "問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例問題文１の回答例",
        answer: "",
        choices: [],
        select: false,
      },
      {
        question_id: 2,
        question:
          "問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２問題文２",
        answer_example:
          "問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例問題文２の回答例",
        answer: "",
        choices: [],
        select: false,
      },
      {
        question_id: 3,
        question:
          "問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３問題文３",
        answer: "選択肢2",
        answer_example: "",
        choices: ["選択肢1", "選択肢2", "選択肢3", "選択肢4"],
        select: true,
      },
    ],
    edit_date: "2023-7-6",
    department: "FR",
  },
];

export default answerEditData;
