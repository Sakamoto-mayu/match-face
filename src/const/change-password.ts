export const dummyPasswords = {
  halfSizeEng: {
    seven: 'abcdefg',
    eight: 'abcdefg',
    nine: 'abcdefg',
  },

  halfSizeNum: {
    seven: '1234567',
    eight: '12345678',
    nine: '123456789',
  },

  halfSizeEngNum: {
    seven: 'abcd123',
    eight: 'abcd1234',
    nine: 'abcde1234',
  },

  halfSizeEngSymb: {
    seven: 'abcdef@',
    eight: 'abcdefg@',
    nine: 'abcdefgh@',
  },
  halfSizeNumSymb: {
    seven: '123456#',
    eight: '1234567#',
    nine: '12345678#',
  },
};

// 記号 . _ / # & % = ~ - + * @ ( ) < > [ ] { } が1つ以上含まれる、かつ半角英数字が含まれる、かつ8文字以上
export const correctPassword = 'newpassword#1';
// 条件に合致しないパスワード
export const incorrectPassword = 'incoreect';
// incorrectPasswordと一致しない,条件に合致しないパスワード
export const anyPassword = 'any';
