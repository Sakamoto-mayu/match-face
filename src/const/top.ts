export const MenuContentsArray = [
  {
    id: 1,
    title: '回答する',
    url: '/questions',
    description: '質問に答えて顔合わせの準備をします。',
    imgUrl: '/icon/document_icon.png',
    imgAlt: '回答一覧画面アイコン',
  },
  {
    id: 2,
    title: '回答履歴',
    url: '/histories',
    description: '自身の回答をここから確認してください。',
    imgUrl: '/icon/history_icon.png',
    imgAlt: '回答履歴画面アイコン',
  },
  //   {
  //     id: 3,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 4,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 5,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 6,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
  //   {
  //     id: 7,
  //     title: '回答履歴',
  //     url: '/histories',
  //     description: '自身の回答をここから確認してください。',
  //     imgUrl: '/icon/history_icon.png',
  //     imgAlt: '回答履歴画面アイコン',
  //   },
];
