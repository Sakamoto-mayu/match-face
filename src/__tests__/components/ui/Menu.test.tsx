import Menu from '@/components/pages/general/top/Menu';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { menuContentsArray } from '@/const/top';

// ダミーデータ
const menuContent = menuContentsArray[0];

describe('スナップショット', () => {
  test('スナップショット', () => {
    const { container } = render(
      <Menu
        url={menuContent.url}
        title={menuContent.title}
        description={menuContent.description}
        imgUrl={menuContent.imgUrl}
        imgAlt={menuContent.imgAlt}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('引数に指定のデータを与えた時', () => {
  beforeEach(() => {
    render(
      <Menu
        url={menuContent.url}
        title={menuContent.title}
        description={menuContent.description}
        imgUrl={menuContent.imgUrl}
        imgAlt={menuContent.imgAlt}
      />
    );
  });
  test('リンクが指定したデータのurlになっていること', () => {
    const menuLink = screen.getByRole('link');
    expect(menuLink).toHaveAttribute('href', `${menuContent.url}`);
  });

  test('見出しが指定したデータのtitleになっていること', () => {
    const menuTitle = screen.getByText(`${menuContent.title}`);
    expect(menuTitle).toBeInTheDocument();
  });

  test('詳細説明が指定したデータのdescriprionになっていること', () => {
    const menuDescription = screen.getByText(`${menuContent.description}`);
    expect(menuDescription).toBeInTheDocument();
  });

  test('画像のurlが指定したデータのimgUrlになっていること', () => {
    const menuImg = screen.getByRole('img');
    // next.jsが自動でencodeを行うため、menuContent.imgUrlをencodeする
    const encodedImgUrl = encodeURIComponent(`${menuContent.imgUrl}`);
    // 完全一致ができなかったため部分一致
    expect(menuImg.getAttribute('src')).toContain(encodedImgUrl);
  });

  test('画像のaltが指定したデータのimgAltになっていること', () => {
    const menuImg = screen.getByRole('img');
    expect(menuImg).toHaveAttribute('alt', `${menuContent.imgAlt}`);
  });
});
