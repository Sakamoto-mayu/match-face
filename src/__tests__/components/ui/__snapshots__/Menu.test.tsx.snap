// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`スナップショット スナップショット 1`] = `
<div>
  <a
    href="/questions"
  >
    <button
      class="flex p-5 w-96 h-40 shadow-lg hover:bg-gray-100 flex-col active:bg-gray-200"
    >
      <p
        class="border-b-[1px] border-b-black shadow-md text-2xl text-left w-full font-bold"
      >
        回答する
      </p>
      <div
        class="flex pt-4 w-full"
      >
        <p
          class="text-left mr-auto"
        >
          質問に答えて顔合わせの準備をします。
        </p>
        <img
          alt="回答一覧画面アイコン"
          class="pt-2 pl-2"
          data-nimg="1"
          decoding="async"
          height="70"
          loading="lazy"
          src="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75"
          srcset="/_next/image?url=%2Ficon%2Fdocument_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fdocument_icon.png&w=256&q=75 2x"
          style="color: transparent;"
          width="70"
        />
      </div>
    </button>
  </a>
</div>
`;
